### ISSUE TITLE
[BUG] thread spin after radius error

### ISSUE THREAD

**Issue by rrevels-bw** [OpenSIPS/opensips#1864]
**OpenSIPS version you are running**

```
version: opensips 2.3.6 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll_lt, epoll_et, sigio_rt, select.
git revision: 46e9eec39
main.c compiled on 14:44:24 Sep 23 2019 with gcc 4.8.5

```

**Describe the bug**
After getting the following error in the opensips log the thread stops handling calls and starts using the max cpu it can get a hold of.

2019-10-16T20:28:13.561169+00:00 proxy /usr/local/opensips/sbin/opensips[20916]: ERROR:aaa_radius:rad_avp_add: invalid value argument
2019-10-16T20:28:13.570476+00:00 proxy /usr/local/opensips/sbin/opensips[20916]: ERROR:acc:acc_aaa_request: failed to add acc-CDR-Version, 8

 PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND                                                                                                                                          
20916 opensips  20   0 1323188 227948 212780 R  99.3  5.9 318:54.93 opensips                                                                                                                                         
20913 opensips  20   0 1323192 227932 212780 S   2.0  5.9 109:49.35 opensips                                                                                                                                         
20919 opensips  20   0 1323192 227952 212792 S   2.0  5.9  80:07.07 opensips                                                                                                                                         

I've left a couple other threads in the process list to show what they would normally use for cpu.

**To Reproduce**
1. Start OpenSIPS
2. Start traffic
3. Run for days or weeks with no issue
4. Get error message in log and find no other issues indicated at all
5. check top and observe error thread using 100% cpu
6. notice thread no longer producing log messages which would be normal in call processing.

**Expected behavior**
1. Not expecting the error message about radius.
2. Not expecting single thread to use a lot of cpu while processing no calls

**Relevant System Logs**

**OS/environment information**
 - Operating System: el7.x86_64
 - OpenSIPS installation: compiled from release version source
 - other relevant information:

**Additional context**

**Comment by rrevels-bw** [comment]
I wonder if this could be related to #1580

**Comment by rrevels-bw** [comment]
I forced a dump on the thread that was hung.  We haven't seen this issue on any other servers since the day that two of them did this within hours of each other.  I am thinking there was an external event that caused some type of memory corruption on the VM.  For what its worth, here is where it was hung.  The thing I found interesting is the error in the logs was about radius values not being available but this backtrace mentions the filename of the file based transaction reporting.  

#1  0x00007fd61b945bee in get_lock (lock=0x7fd623534020) at ../../evi/../fastlock.h:221
        i = 0
#2  acc_db_request (rq=rq@entry=0x7fd6275acc28, rpl=<optimized out>, ins_list=ins_list@entry=0x7fd61bb85ef0 <acc_ins_list>, cdr_flag=cdr_flag@entry=0) at acc.c:614
        my_ps_ins = 0x0
        my_ps_ins2 = 0x0
        my_ps_ins3 = 0x0
        my_ps = 0x0
        my_ps2 = 0x0
        my_ps3 = 0x0
        m = <optimized out>
        n = <optimized out>
        i = 6
        j = <optimized out>
        _created = <optimized out>
        _setup_time = <optimized out>
        extra_start = 7
        extra = <optimized out>
        ctx = <optimized out>
        __FUNCTION__ = "acc_db_request"
#3  0x00007fd61b94f3a5 in acc_onreply (ctx=0x7fd623534020, code=<optimized out>, reply=<optimized out>, req=0x7fd6275acc28, t=<optimized out>) at acc_logic.c:830
        dst_uri_bk = {s = 0x0, len = 0}
        dlg = <optimized out>
        ctx_s = {s = 0x0, len = 6275835}
        flags = 0x7fd623534040
        new_uri_bk = {s = 0x7fd62379b4c8 "sip:+19198788990@192.168.0.117", len = 31}
        table = {s = 0x7fd62974b058 "trans13g\030", len = 7}
#4  tmcb_func (t=<optimized out>, type=<optimized out>, ps=<optimized out>) at acc_logic.c:1027
        ctx = 0x7fd623534020
#5  0x00007fd61fb1b6c9 in run_trans_callbacks (type=type@entry=128, trans=trans@entry=0x7fd625234400, req=req@entry=0x7fd6275acc28, rpl=<optimized out>, code=<optimized out>) at t_hooks.c:209
        params = {req = 0x7fd6275acc28, rpl = 0x7fd660bcb1e8, code = 200, param = 0x7fd627ae83b8, extra1 = 0x7ffe217e6b20, extra2 = 0x7fd625234518}

**Comment by bogdan-iancu** [comment]
@rrevels-bw , actually there is a bug there - checking the code, the culprit lock (that you noticed with gdb) is left un-released in case of an "RADIUS AVP add " failure.

**Comment by bogdan-iancu** [comment]
@rrevels-bw , I fixed the issue in head, 3.0 and 2.4 (the maintained versions). Your 2.3 is not officially maintained anymore , but you can try to cherry-pick the fix from 2.4 ;)

**Comment by rrevels-bw** [comment]
That is good timing as I had just yesterday pivoted back to working with the 2.4 branch.  A lot of the segfaults I was seeing are gone but I still have one (that ive seen so far) left and will open a crash report on it.

Thank you! for following up on this one and finding/fixing that issue.
